### PR TITLE
[WIP] Vertex Array Objects (VAO) support

### PR DESCRIPTION
**Pull request by takahirox**
This PR introduces Vertex Array Objects (VAO) into Three.js.

I'll add comment here and inline later. Please wait for a little while...

Feedbacks and bug reports are welcome.

Inspired by #14224

/cc @Mugen87

### CODE REVIEW

File: src/renderers/webgl/WebGLBindingStates.js [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+				currentState.program = program.id;
+				currentState.wireframe = wireframe;
+
+				updateBuffers = true;
+
+			}
+
+		}
+
+		if ( updateBuffers ) {
```
[the remark this answers is not in the document]
Comment (Mugen87) [reply src/renderers/webgl/WebGLBindingStates.js]: Changing the data of an attribute (like updating the data of the position attribute) is unrelated to the `VAO`.

[the remark this answers is not in the document]
Comment (DavidPeicho) [reply src/renderers/webgl/WebGLBindingStates.js]: Changing the buffer data no, but changing the stride/component/offset is if I am correct

**Comment by takahirox** [comment]
Sorry, let me make a new PR. I did misoperation...

File: src/renderers/webgl/WebGLBindingStates.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (takahirox) [reply src/renderers/webgl/WebGLBindingStates.js]: That's one of the things I wanna discuss. I'll add comment about it in the new PR.

**Pull request closed without merging**